apsuite/optics_analysis/base_correction.py

```diff
@@ -0,0 +1,188 @@
+        self.method = method
+        self.grouping = grouping
+
+    @property
+    def grouping(self):
+        """."""
+        return self._grouping
+
+    @grouping.setter
+    def grouping(self, value):
+        if value is None:
+            return
+        elif isinstance(value, str):
+            self._grouping = int(value in BaseCorr.GROUPING._fields[1])
+        elif int(value) in BaseCorr.GROUPING:
+            self._grouping = int(value)
+
+    @property
+    def grouping_str(self):
+        """."""
+        return BaseCorr.GROUPING._fields[self._grouping]
+
+    @property
+    def method(self):
+        """."""
+        return self._method
+
+    @method.setter
+    def method(self, value):
+        if value is None:
+            return
+        elif isinstance(value, str):
+            self._method = int(value in BaseCorr.METHODS._fields[1])
+        elif int(value) in BaseCorr.METHODS:
+            self._method = int(value)
+
+    @property
+    def method_str(self):
+        """."""
+        return BaseCorr.METHODS._fields[self._method]
+
+    def define_knobs(self, f_knobs, d_knobs, strength_type):
+        """."""
+        self.knobs = KnobTypes(Focusing=f_knobs, Defocusing=d_knobs)
+        self.strength_type = strength_type
+
+    def get_parameter(self, model=None):
+        """."""
+        raise NotImplementedError
+
+    def calc_jacobian_matrix(self, model=None):
+        """."""
+        raise NotImplementedError
+
+    def get_strength(self, model=None, knobs=None):
+        """."""
+        if model is None:
+            model = self.model
+        if knobs is None:
+            knobs = self.knobs.ALL
+        stren = []
+        for knb in knobs:
+            stren_mag = []
+            for mag in self.fam[knb]['index']:
+                stren_seg = []
+                for seg in mag:
+                    stren_seg.append(getattr(model[seg], self.strength_type))
+                stren_mag.append(sum(stren_seg))
+            stren.append(np.mean(stren_mag))
+        return np.array(stren)
+
+    def correct_parameters(self,
+                           goal_parameters,
+                           model=None,
+                           jacobian_matrix=None,
+                           tol=1e-6,
+                           nr_max=10,
+                           nsv=None):
+        """."""
+        if model is None:
+            model = self.model
+        if self.method not in BaseCorr.METHODS:
+            raise Exception('Invalid correction method!')
+
+        if jacobian_matrix is None:
+            jmat = self.calc_jacobian_matrix(model)
+        else:
+            jmat = _dcopy(jacobian_matrix)
+
+        nominal_stren = self.get_strength(model)
+        if self._method == BaseCorr.METHODS.Proportional:
+            jmat *= nominal_stren
+        if self._grouping == BaseCorr.GROUPING.TwoKnobs:
+            jmat = self._group_2knobs_matrix(jmat)
+
+        U, S, V = np.linalg.svd(jmat, full_matrices=False)
```

Comment:
Pylint requires variable names with three or more letters.
In other python modules I used `umat`, `smat`, `vmat`... I did not like these names but this is what I come up compatible with pylint. Do you have suggestions? Whatever names we choose I vote for using the same names across python modules for variables which represent the same idea. I think this *straitjacket* makes code maintenance much easier for original and new-comer developers.

Reply:
I think `umat`, `smat`, `vmat` is ok. I changed here and I will standardize in other modules which uses SVD.